firestore-bigquery-export/scripts/gen-schema-view/src/util.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as glob from "glob";
+import * as path from "path";
+
+import { FirestoreSchema } from "./schema";
+
+import { existsSync, readdirSync, lstatSync } from "fs";
+
+export function readSchemas(
+  globs: string[]
+): { [schemaName: string]: FirestoreSchema } {
+  let schemas = {};
+  let expanded = expandGlobs(globs);
+  for (var i = 0; i < expanded.length; i++) {
+    let dirent = resolveFilePath(expanded[i]);
+    let stats = lstatSync(dirent);
+    if (stats.isDirectory()) {
+      let directorySchemas = readSchemasFromDirectory(dirent);
+      for (let schemaName in directorySchemas) {
+        if (schemas.hasOwnProperty(schemaName)) {
+          warnDuplicateSchemaName(schemaName);
+        }
+        schemas[schemaName] = directorySchemas[schemaName];
+      }
+    } else {
+      let schemaName = filePathToSchemaName(dirent);
+      if (schemas.hasOwnProperty(schemaName)) {
+        warnDuplicateSchemaName(schemaName);
+      }
+      schemas[schemaName] = readSchemaFromFile(dirent);
+    }
+  }
+  return schemas;
+}
+
+function warnDuplicateSchemaName(schemaName: string) {
+  console.log(
+    `Found multiple schema files named ${schemaName}! Only the last one will be created!`
```

Comment:
Maybe change last sentence to:

Only the most recent one will be used to create the schema view!

Reply:
The word "recent" implies that we are somehow iterating over these in a temporal order, but we aren't. @fredzqm left a comment about this requesting the opposite of what you're asking. I like adding "will be used to create the schema view" though.